src/lightning/pytorch/utilities/exceptions.py

```diff
@@ -27,3 +29,9 @@ class SIGTERMException(SystemExit):
 
 class _TunerExitException(Exception):
     """Exception used to exit early while tuning."""
+
+
+def _replace_message(exception: BaseException, pattern: str, new_message: str) -> None:
+    exception.args = tuple(
```

Comment:
Did you explore playing with the `__cause__` attribute?

https://legacy.python.org/dev/peps/pep-3134/
https://stackoverflow.com/questions/24752395/python-raise-from-usage

We should keep the original message too. As `raise ... from ...` does

Reply:
Are you suggesting to return a new exception from the on_exception method?

Reply:
No. I was just asking if you had tried hacking around the `__cause__` attribute. I don't know if this is possible.

This would also be replaced by https://peps.python.org/pep-0678/ when 3.11 is our minimum version. Can you add a note about that?

Reply:
I unfortunately don't see how I could manipulate `__cause__`. The value of `__cause__` needs to be another exception.

Reply:
And what's the issue with adding another exception?

```python
try:
    raise ValueError("foobar")
except BaseException as err:
    if _PYTHON_GREATER_EQUAL_3_11:
        err.add_note("barfoo")
    else:
        err.__cause__ = type(err)("barfoo")
    raise
```

Reply:
That's what I tried. If you look at the ddp_spawn test I added, there the exception type is ProcessRaisedException, and it can't be instantiated with just `type(err)(*exception.args)`. It just doesn't generalize like that.

Reply:
It requires #14652 before it can be properly tested.